Sources/DistributedActors/DistributedActors.docc/ClusterSingleton.md

```diff
@@ -0,0 +1,178 @@
+<!-- Enable additional code snippet verification; all inline snippets concat into a "compile test" and verified when built with VALIDATE_DOCS=1 -->
+
+
+### Implementing a Cluster Singleton
+
+A cluster singleton is a distributed actor using the ``ClusterSystem`` actor system and conforming to the ``ClusterSingleton`` protocol.
+
+We can implement a singleton by declaring a `distributed actor` as follows:
+
+```swift
+import DistributedActors
+typealias DefaultDistributedActorSystem = ClusterSystem
+```
+
+```swift
+distributed actor PrimaryOverseer: ClusterSingleton {
+  var status = WorkStatus()
+
+  distributed func workStatus() -> WorkStatus { 
+    self.status
+  }
+}
+
+distributed actor Worker { /* ... */ }
+
+struct WorkStatus: Codable { 
+  var workerCapacity: Int = 0
+  var workInProgress: Int = 0
+}
+``` 
+
+The protocol does not have any protocol requirements that are required to be implemented explicitly, however it allows the runtime to inform the singleton instance about important events regarding its lifecycle.
+
+
+### Obtaining Cluster Singleton references
+
+To use singletons in your system you must enable the singleton plugin first:
+
+```swift
+let system = await ClusterSystem("SingletonExample") { settings in
+  settings.plugins.install(plugin: ClusterSingletonPlugin())
+}
+```
+
+Next, you'll be able to make use of the `.singleton` property on the cluster system to perform various actions on the active plugin.
+Most importantly, you can issue a `host` call, in order to inform the singleton boss that this node is capable of hosting the given singleton:
+
+```swift
+let overseerSingleton: PrimaryOverseer =    
+  try await system.singleton.host(name: "overseer") { actorSystem in 
+    PrimaryOverseer(actorSystem: actorSystem)
+  }
+```
+
+The passed `name` is the unique name that the singleton shall be identified with in the cluster. In other words, since we've given this singleton the name `"overseer"`, there will be always at-most-one `"overseer"` actor instance active in the cluster. This allows us to have multiple singletons of the same actor type, but different identities. For example, we could have an `"simple-work-overseer"` and an `"complicated-work-overseer"` singletons running side-by-side. Conceptually, they are different singletons after all, and we can be using them differently. 
+
+A `PrimaryOverseer` obtained this way is actually a proxy that will redirect calls made on it to wherever the actual singleton instance is currently hosted. 
+This includes the local cluster member, in case that is where the singleton ends up _allocated_ (see: <doc#. From the perspective of the caller though, the location of the singleton remains transparent - with the benefit that if the calls are actually remote, they don't have to cross a network boundary and will be more efficient, we should not rely on this though, and program against the singleton as-if it might be remote since it's location might be changing at runtime.
```

Comment:
```suggestion
This includes the local cluster member, in case that is where the singleton ends up _allocated_ (see: <doc#. From the perspective of the caller though, the location of the singleton remains transparent - with the benefit that if the calls are actually remote, they don't have to cross a network boundary and will be more efficient. We should not rely on this though, and we should program against the singleton as-if it might be remote since its location might be changing at runtime.
```